File: src/main/java/com/innercircle/survey/model/Answer.java
package com.innercircle.survey.model;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED
)
@Entity
public class Answer extends BaseEntity {
    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    private Question question;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "response_id")
    private Response response;

    private Integer questionOrder;

    @Enumerated(EnumType.STRING)
    private QuestionType questionType;

    private String answerValue;

    private Integer answerSingleOption;

    private String answerMultipleOptions;

    private Integer version;

    @Builder
    private Answer(Question question, Response response, Integer questionOrder, QuestionType questionType, String answerValue, String answerMultipleOptions, Integer answerSingleOption, Integer version) {
        this.question = question;
        this.response = response;
        this.questionOrder = questionOrder;
        this.questionType = questionType;
        this.answerValue = answerValue;
        this.answerSingleOption = answerSingleOption;
        this.answerMultipleOptions = answerMultipleOptions;
        this.version = version;
    }
}

File: src/main/kotlin/org/inner/circle/survey/domain/Answer.kt
package org.inner.circle.survey.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "answer")
class Answer(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_response_id", nullable = false)
    var surveyResponse: SurveyResponse,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id", nullable = false)
    var question: Question,
    @Column(nullable = false)
    var content: String,
) {
    constructor() : this(surveyResponse = SurveyResponse(), question = Question(), content = "")
}
